packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -0,0 +1,269 @@
+
+        if (refreshToken) {
+            console.log('Using refresh token to get new access token.')
+            // TODO: error handling
+            const res = await helpers.refreshAccessToken(this.client, {refreshToken})
+            this.handleTokenResponse(res, true)
+            return this.data
+        }
+
+        // TODO: error handling
+        const res = await helpers.loginGuestUser(this.client, {redirectURI: this.redirectURI})
+        this.handleTokenResponse(res, true)
+
+        return this.data
+    }
+
+    private handleTokenResponse(res: ShopperLoginTypes.TokenResponse, isGuest: boolean) {
+        const refreshTokenKey = isGuest ? 'refreshTokenGuest' : 'refreshTokenRegistered'
+        this.set('accessToken', `Bearer ${res.access_token}`)
+        this.set(refreshTokenKey, res.refresh_token, {
+            expires: this.REFRESH_TOKEN_EXPIRATION_DAYS,
+        })
+    }
+
+    /**
+     * The ready function returns a promise, signaling whether or not
+     * the access token is avaliable. If access token is not avaliable
+     * this method will try to re-initialize.
+     *
+     * We use this method to block those commerce api calls that
+     * requires an access token.
+     */
+    ready() {
+        console.log('ready')
+        if (!this.pending) {
+            console.log('no ready promise, initializing')
+            this.pending = this.init()
+        }
+        return this.pending
+    }
+
+    async loginRegisteredUserB2C(credentials: Parameters<Helpers['loginRegisteredUserB2C']>[1]) {
+        const redirectURI = this.redirectURI
+        const usid = this.get('usid')
+        const request = async () => {
+            const res = await helpers.loginRegisteredUserB2C(this.client, credentials, {
+                redirectURI,
+                ...(usid && {usid}),
+            })
+            await this.handleTokenResponse(res, true)
+            return this.data
+        }
+        this.pending = request()
+        return this.pending
+    }
+
+    async logout() {
+        const request = async () => {
+            const res = await helpers.logout(this.client, {
+                refreshToken: this.get('refreshTokenRegistered'),
+            })
+            await this.handleTokenResponse(res, true)
+            return this.data
+        }
+        this.pending = request()
+        return this.pending
+    }
+}
+
+export default Auth
+
+type ArgWithHeaders =
+    | {
+          parameters?: any
+          headers?: Record<string, string>
+      }
+    | undefined
+
+/**
+ * A ultility function to inject access token as Authorization header
+ * into the commerce-sdk-isomorphic methods' first argument.
+ *
+ * @Internal
+ */
+export const withAccessToken = <T extends ArgWithHeaders>(arg: T, accessToken: string) => {
+    return {
+        ...arg,
+        headers: {
+            Authorization: accessToken,
+            ...arg?.headers,
```

Comment:
To ensure the `Authorization` won't be overridden by the spreading from arg in case it exists

```suggestion
            ...arg?.headers,
            Authorization: accessToken,
```

Reply:
I'm thinking to allow people to override Authorization header...